src/cysignals/implementation.c

```diff
@@ -222,10 +286,26 @@ static void cysigs_interrupt_handler(int sig)
  * Outside of sig_on(), we terminate Python. */
 static void cysigs_signal_handler(int sig)
 {
+#ifndef POSIX
+    /*
+     * Since we are using ANSI signals, we must reset the handler.
+     */
+    if (signal(sig, cysigs_signal_handler) == SIG_ERR)
+    {
+        perror("signal");
+        exit(1);
+    }
+#endif
+
+#ifdef POSIX
     sig_atomic_t inside = cysigs.inside_signal_handler;
     cysigs.inside_signal_handler = 1;
 
     if (inside == 0 && cysigs.sig_on_count > 0 && sig != SIGQUIT)
+#else
+    int inside = 0; // not used for windows
+    if(cysigs.sig_on_count > 0)
```

Comment:
Is there a particular reason that you are not using `inside_signal_handler` on Windows?